salt/salt.spec

```diff
@@ -829,6 +830,7 @@ than serially.
 Summary:        The client component for Saltstack
 Group:          System/Management
 Requires:       %{name} = %{version}-%{release}
+Requires:       (%{name}-transactional-update = %{version}-%{release} if read-only-root-fs)
```

Comment:
I think you need to guard this with a OS version which know this construct.

What will happen when this run on SLES12, RHEL7 or RHEL8 ?

Reply:
Indeed. I added `%if 0%{?suse_version} > 1500 || 0%{?sle_version} > 1500`, I hope that this include openSUSE MicroOS and SLE Micro